goal_src/jak1/kernel/gcommon.gc

```diff
@@ -294,9 +294,16 @@
   `((the (current-method-function-type) (find-parent-method (current-method-type) (current-method-id))) ,@args))
 
 (defmacro as-type (this type)
-  "Macro to _safely_ convert to a different type, returning #f if the type doesn't match.
-   Does a runtime type check so it's expensive."
-  `(if (and (nonzero? ,this) (type-type? (-> ,this type) ,type)) (the-as ,type ,this)))
+  "If this is the specified type, return this cast to that type. Otherwise, return #f."
+  (with-gensyms (obj)
+    `(the ,type (let ((,obj ,this))
+                  (if (type? ,obj ,type)
```

Comment:
Jak 1 doesn't have `type?`, only `type-type?`.

Reply:
oops - I should have left this. The decompiler doens't recognize the jak 1 version yet.